app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageViewModel.kt

```diff
@@ -730,31 +758,52 @@ class NewMessageViewModel @Inject constructor(
         }.onFailure(Sentry::captureException)
     }
 
-    fun executeDraftActionWhenStopping(
-        action: DraftAction,
-        isFinishing: Boolean,
-        isTaskRoot: Boolean,
-        subjectValue: String,
-        uiBodyValue: String,
-        startWorkerCallback: () -> Unit,
-    ) = globalCoroutineScope.launch(ioDispatcher) {
+    fun saveBodyAndSubject(subject: String, html: String) {
+        globalCoroutineScope.launch(ioDispatcher) {
+            _subjectAndBodyChannel.send(Triple(subject, html, channelExpirationIdTarget))
+        }
+    }
 
-        val localUuid = draftLocalUuid ?: return@launch
+    fun discardOldBodyAndSubjectChannelMessages() {
+        channelExpirationIdTarget++
+    }
+
+    fun waitForBodyAndSubjectToExecuteDraftAction(draftSaveConfiguration: DraftSaveConfiguration) {
+        subjectAndBodyJob?.cancel()
+        subjectAndBodyJob = globalCoroutineScope.launch(ioDispatcher) {
+            val subject: String
+            val body: String
+            while (true) {
```

Comment:
Isn't explaining that receive is blocking the same as explaining that `val` instances can only be read and not set? If you see what I mean. Shouldn't we be careful about over explaining things? 

I think that small comments like this one really can't hurt so I added it anyway. But I raised the question regardless to open the discussion if you have an opinion about it

Reply:
This is a bit of an over explaination indeed, but I think it can be needed as neither of us apart from Siramd had heard of the `ReceiveChannel` before because it's not that common to use it, and the fact that even when reading the documentation, I didn't quickly understood that it was blocking.